### PR TITLE
chore: fix docs typo

### PR DESCRIPTION
**Pull request by yyzxw**
**Please provide a description of this PR:**

fix some docs typo, delete outdated links.

### CODE REVIEW

File: pilot/ARCHITECTURE.md [hunk 1]
```diff
@@ -7,8 +7,8 @@ Istiod is structured as a modular monolith, housing a wide range of functionalit
 
 Istiod's primary role - and most code - is to dynamically configure proxies (Envoy sidecars and ingress, gRPC, ztunnel, and more). This roughly consists of 3 parts:
 1. Config ingestion (inputs to the system)
-1. Config translation
-1. Config serving (XDS)
+2. Config translation
```
Comment (hanxiaop) [inline pilot/ARCHITECTURE.md:10]: I don't get this. Isn't the number automatically formatted?

**Review by howardjohn — COMMENTED** [review]
Please focus on https://github.com/istio/istio/issues?q=is%3Aopen+is%3Aissue+label%3A%22community%2Fgood+first+issue%22 type issues

File: pilot/ARCHITECTURE.md [hunk 2]
```diff
@@ -296,11 +296,11 @@ The Credentials controller exposes access to TLS certificate information, stored
 
 The Discovery Filter controller is used to implement the `discoverySelectors` field of `MeshConfig`. This controller reads `Namespace`s in the cluster to determine if they should be "selected". Many controllers consumer this filter to only process a subset of configurations.
 
-#### Multicluster
```
[the remark this answers is not in the document]
Comment (frankbu) [reply pilot/ARCHITECTURE.md:299]: and in the glossary https://istio.io/latest/docs/reference/glossary/#multicluster

[the remark this answers is not in the document]
Comment (yyzxw) [reply pilot/ARCHITECTURE.md]: yes, it's reverted.

File: pilot/ARCHITECTURE.md (hunk 1 above)
Comment (yyzxw) [reply pilot/ARCHITECTURE.md:10]: Its numbers are equal and I want to keep it incremental.

Comment (ymesika) [reply pilot/ARCHITECTURE.md:10]: @yyzxw it's valid Markdown numbering method to avoid re-numbering all the following items if you insert new item in the middle.

File: manifests/charts/UPDATING-CHARTS.md [hunk 3]
```diff
@@ -55,7 +55,6 @@ This will regenerate the Go structs used for schema validation.
 ## Step 4. Update the generated manifests
 
 Tests of istioctl use the auto-generated manifests to ensure that the istioctl binary has the correct version of the charts.
-These manifests can be found in [gen-istio.yaml](../charts/istio-control/istio-discovery/files/gen-istio.yaml).
 To regenerate the manifests, run:
```
Comment (ymesika) [inline manifests/charts/UPDATING-CHARTS.md:58]: Why did you remove it?

File: samples/extauthz/README.md [hunk 4]
```diff
@@ -19,7 +19,7 @@ Note that `a` is just a default value for testing. It can be changed with the fl
 
     Note, you can also deploy the Ext Authz service locally with the application container in the same pod, see the example in `local-ext-authz.yaml`.
 
-1. Verify the Ext Authz server is up and running:
+2. Verify the Ext Authz server is up and running:
```
Comment (ymesika) [inline samples/extauthz/README.md:22]: ditto

**Review by ymesika — CHANGES_REQUESTED** [review]
Few markdown list items re-numbering that you've made are not necessary.
Also left a comment about a removed text which I'm not sure why.

File: pilot/ARCHITECTURE.md (hunk 1 above)
Comment (dhawton) [reply pilot/ARCHITECTURE.md:10]: The numbering is by default supposed to be `1.` that is why lint is failing. This is the rule being applied here: https://github.com/markdownlint/markdownlint/blob/main/docs/RULES.md#md029---ordered-list-item-prefix

Keep them all `1.`

**Review by dhawton — CHANGES_REQUESTED** [review]

File: manifests/charts/UPDATING-CHARTS.md (hunk 3 above)
Comment (yyzxw) [reply manifests/charts/UPDATING-CHARTS.md:58]: the file has been removed(https://github.com/istio/istio/blob/a04ccfde0f984dd0b17a4298acdddb30c0e850f8/releasenotes/notes/drop-kustomize.yaml#L8),   or has it been moved somewhere?

File: pilot/ARCHITECTURE.md (hunk 1 above)
Comment (yyzxw) [reply pilot/ARCHITECTURE.md]: Thank you all, I've reverted.

File: manifests/charts/UPDATING-CHARTS.md (hunk 3 above)
Comment (ymesika) [reply manifests/charts/UPDATING-CHARTS.md:58]: You are right.

**Review by ymesika — APPROVED** [review]
lgtm

**Review by dhawton — APPROVED** [review]

**Review by hanxiaop — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]

File: samples/httpbin/README.md [hunk 5]
```diff
@@ -37,5 +37,5 @@ time kubectl run -i --rm --restart=Never dummy --image=dockerqa/curl:ubuntu-trus
 
 You can also test the httpbin service by starting the [sleep service](../sleep) and calling httpbin from it.
 
-A third option is to access the service from outside of the mesh through an Ingress Gateway.
+A third option is to access the service from the outside of the mesh through an Ingress Gateway.
```
Comment (frankbu) [inline samples/httpbin/README.md:40]:
```suggestion
A third option is to access the service from outside of the mesh through an Ingress Gateway.
```

Comment (ericvn) [inline samples/httpbin/README.md:40]: or to jive more with your earlier change:
```suggestion
A third option is to access the service from outside the mesh through an Ingress Gateway.
```
